### ISSUE TITLE
Push V5 documentation and additions

### ISSUE THREAD

**Issue by Manduro** [driftyco/ng-cordova#1125]
The new phonegap-plugin-push is supported, but it's not documented anywhere.

There are some new features to implement as well, as listed on https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/API.md:
- PushNotification.hasPermission(successHandler)
- push.off(event, callback)
- push.getApplicationIconBadgeNumber
- push.finish(successHandler, errorHandler) (PR #1091 implements this)

The implementation could be simplified a little as per PR #1052.

**Comment by JaKXz** [comment]
+1 - thanks for trying out the new plugin!

**Comment by felipecaparelli** [comment]
I am trying to use this new plugin too... it register the device token, but it doesn't receives any notification from GCM (listeners does not work propertly). Follow my code:

`$cordovaPushV5.register().then(function(token){
                            console.log('$cordovaPushV5:REGISTERED', token);
                            //it cames here... but...
                            $cordovaPushV5.onError();
                            $cordovaPushV5.onNotification();
                        }, function(err){
                            console.error('$cordovaPushV5:REGISTER_ERROR', err);
                        });

                        $rootScope.$on('$cordovaPushV5:notificationReceived', function (event, notification) {
                            switch( e.event )
                            {
                            case 'registered':
                                if ( e.regid.length > 0 ) {
                                    console.log("regID = " + e.regid);
                                }
                            break;

                            case 'message':
                                if ( e.foreground ) {
                                    console.log("foreground");
                                } else {  // otherwise we were launched because the user touched a notification in the notification tray.
                                    console.log("not foreground");
                                }
                            break;

                            case 'error':
                                console.log("gcm error ");
                            break;

                            default:
                                console.log("evento desconhecido ");
                            break;
                          }
                        });

                        $rootScope.$on('$cordovaPushV5:errorOccurred', function (event, error) {
                                console.log("gcm error ");
                                console.log(error);
                        });`

**Comment by HowardLoTW** [comment]
Hello,  +1, too
 
I think this is quit important since the deprecated pushPlugin doesn't support changing notification icon.
In order to do this, we should use the non-deprecated push plugin.
But the push notification doc hasn't been updated. 

Is someone trying to use this new push plugin ? 
I'm also trying this, too. But without any docs, it'll be difficult for new comers to this new push plugin.

If there's any help needed, please let me know.

Thanks.

**Comment by HowardLoTW** [comment]
Hello, @felipecaparelli 
Have you tried the new plugin successfully ? Thanks.

**Comment by kashyapmukkamala** [comment]
@HowardLoTW @felipecaparelli Were you able to get this to work? I am getting notifications but I am unable to navigate/take action by clicking on the notification on the lock screen or the notifications tray. It just takes me to the home page of my app. Is there a way I can trigger some callback and know which notification the user tapped on? I have checked the entire source code of cordovaPushV5 but couldnt find anything

**Comment by HowardLoTW** [comment]
@kashyapmukkamala If you want to get your users to know which notifications is tapped.
You can add payload data to notification you sent from the server side ( For me, I use node-gcm, and there's a payload field specified when the notification is sent to the device. ) Then you can get the payload data from $cordovaPushV5:notificationReceived notification object, which let you know what action should be performed when user tapped the notification.

**Comment by HowardLoTW** [comment]
@kashyapmukkamala Here you can take a look my code 👍 

Ionic 
```
		/*
		 * Non deprecated version of Push notification events
		 */
	    function registerV5() {

	    	$ionicLoading.show({
		    	template: '<ion-spinner></ion-spinner>'
		    });

            if (ionic.Platform.is('browser')) {
                alert("You are running on broswer, please switch to your device. Otherwise you won't get notifications");
                $ionicLoading.hide();
                return;
            }

            /**
             * Configuration doc:
             * https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/API.md#pushnotificationinitoptions
             */
            var GCM_PROJECT_ID = 'xxxxxx';

            $cordovaPushV5.initialize({
                "android": {
                	"clearNotifications": false,
                    "senderID" : GCM_PROJECT_ID
                }
            });

            $cordovaPushV5.register().then(function (deviceToken) {
                console.log("Successfully registered", deviceToken);
                
                $scope.data.deviceToken = deviceToken;

                // Below code required to configure $cordovaPushV5 notifications emitter. 
                // Don't pass function it's not handler.
                $cordovaPushV5.onNotification();
                $cordovaPushV5.onError();

                $ionicLoading.hide();
            }, function (error) {
                console.log("Failed to registered");
                console.log("error object : ", error);
                $ionicLoading.hide();
            });
        }

        $rootScope.$on('$cordovaPushV5:notificationReceived', function(event, data) {
            console.log("notification received");
            console.log("data object: ", data);

            var foreground = data.additionalData.foreground || false;
            var threadID = data.additionalData.payload.threadID || '';
            var group = data.additionalData.payload.group || false;

            if (foreground) {
                // Do something if the app is in foreground while receiving to push - handle in app push handling
                console.log('Receive notification in foreground');
            } else {
                // Handle push messages while app is in background or not started
                console.log('Receive notification in background');
                // Open FB messanger app when user clicks notification UI when app is in background.
				if (typeof data.additionalData.coldstart != "undefined" && data.additionalData.coldstart == false)
					if (!group)
						// Open FB Messenger of specific user chat window
						window.open('fb-messenger://user/' + threadID, '_system', 'location=no');
					else
						// Open FB Messenger of specific group chat window
						window.open('fb-messenger://groupthreadfbid/' + threadID, '_system', 'location=no');                
            }
        });

        $rootScope.$on('$cordovaPushV5:errorOccurred', function(event, error) {
            console.log("notification error occured");
            console.log("event object: ", event);
            console.log("error object: ", error);
        });

```

Nodejs
```
var notificationCount = 0;
function sendNotification(notification) {

    notificationCount = (notificationCount + 1) % 1000000;

    var retry_times = 4;
    var sender = new gcm.Sender(GCM_API_KEY);
    // Follow with API specification from "phonegap-plugin-push" repo
    // https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#priority-in-notifications
    var message = new gcm.Message({
        "data" : {
            "notId" : notificationCount, // phonegap-plugin-push doesn't support string ID.
            "title" : notification.title || "Emotion Push",
            "image" : notification.image || "icon", // The image file name in drawable folders in Android. It can also be a url here.
            "message" : notification.message || "Emotion Push",
            "priority" : 2, // Highest priority.
            "vibrationPattern" : [300, 150, 300], // Vibrate for 300ms then wait 150ms and then vibrate for 300ms.
            "content-available": "1", // Receive notification in background.
            "payload" : { // Payload for client processing.
                "threadID": notification.threadID,
                "group": notification.group
            }
        }
    });

    sender.send(message, [notification.deviceToken], retry_times, function (err, response) {

        console.log('+----------------------------------------------------+');
        console.log('Push notification: ', notification);
        if (err) console.error('Error', err);
        else     console.log('Response', response);
        console.log('+----------------------------------------------------+');
    });
}

```

**Comment by Manduro** [comment]
Fixed by #1279
